btree/btree.c

```diff
@@ -1812,15 +1812,24 @@ static int64_t node_get(struct node_op *op, struct td *tree,
 	 * implemented
 	 * */
 	nt = &fixed_format;
-	op->no_node = nt->nt_opaque_get(addr);
-
 	/**
 	 * TODO: Adding list_lock to protect from multiple threads
 	 * accessing the same node descriptor concurrently.
 	 * Replace it with a different global lock once hash
 	 * functionality is implemented.
 	 */
 	m0_rwlock_write_lock(&list_lock);
+	/**
+	 * If the node was deleted before node_get can acquire list_lock, then
+	 * restart the tick funcions.
+	 */
+	if (!nt->nt_isvalid(addr)) {
+		op->no_op.o_sm.sm_rc = ECHILD;
```

Comment:
Please check if error code should be  M0_ERR(-ECHILD) format instead of ECHILD. Please make similar changes at line 1837

Reply:
Ok.